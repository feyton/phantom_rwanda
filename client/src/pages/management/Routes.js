import React, { useEffect, useState } from 'react';
import { toast } from 'react-toastify';
import TableSkeleton from '../../components/SkeletonUIs/TableSkeleton.js';
import { axiosBase as axios } from '../../utils/Api.js';
import RouteTable from '../routes/components/RouteTable.js';

const Routes = () => {
  const [routes, setroutes] = useState([]);
  const [loading, setloading] = useState(false);
  useEffect(() => {
    const fetchData = async () => {
      try {
        setloading(true);
        const response = await axios.get('/routes');
        setroutes(response.data.data);
      } catch (error) {
        toast(error.message);
      } finally {
        setloading(false);
      }
    };
    fetchData();
  }, []);
  return (
    <div>{!loading ? <RouteTable data={routes} /> : <TableSkeleton />}</div>
  );
};

export default Routes;
