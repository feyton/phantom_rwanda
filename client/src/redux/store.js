import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
