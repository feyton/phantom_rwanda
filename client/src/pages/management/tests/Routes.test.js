import mockAxios from 'axios';
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../../redux/store.js';
import Routes from '../Routes.js';

describe('Route component', () => {
  describe('Should render the Routes components', () => {
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.reject({
        status: 400,
        data: { data: { message: 'Email not found' } }
      });
    });
    const elem = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Routes />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(elem).toMatchSnapshot();
  });

  it('should render the RouteTable component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Routes />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find('RouteTable').length).toEqual(0);
  });
});
