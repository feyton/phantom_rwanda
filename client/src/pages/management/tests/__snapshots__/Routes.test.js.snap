// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <section
    className="flex justify-between content-center pb-7 font-raleway"
  >
    <div>
      <input
        className="p-1 rounded-md border-2"
        name="search"
        onChange={[Function]}
        placeholder="Search routes..."
        type="text"
        value=""
      />
    </div>
    
  </section>
  <table
    className="w-full"
    role="table"
  >
    <thead>
      <tr
        className="pb-10 "
        role="row"
      >
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Destination 1
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Destination 2
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Distance
        </th>
        <th
          className="text-left font-extrabold pr-5 border-b-2 pb-4 font-raleway"
          colSpan={1}
          role="columnheader"
        >
          Management
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    />
  </table>
</div>
`;
